src/main/react/webpack_registration/web_pack/components/pages/FormRegister/Form.js

```diff
@@ -0,0 +1,190 @@
+import React, { useState, useEffect } from "react";
+import { Redirect } from "react-router-dom";
+import Input from "./inputForForm/InputForForm";
+import { dataLogin, dataRegister, errorsServer } from "./inputBase";
+import { RegisterValidation } from "./validationInput";
+import RenderError from "../renderError/RenderError";
+import { Button } from "./button/Button";
+import Extra from "./extra/Extra";
+import useFetch from "../hooks/doFetch";
+import useLocalStorage from "../hooks/useLocalStorage";
+
+import "./form.scss";
+
+const Form = props => {
+  const isLogin = props.match.path === "/registration";
+  let arrayValueForm = isLogin ? dataLogin : dataRegister;
+  const oneClass = isLogin ? "login" : "register";
+
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [readyToEnter, setReadyToEnter] = useState(false);
+
+  const [isChecking, setIsChecking] = useState(false);
+
+  const [loginInputs, setLoginInputs] = useState({});
+  const [registerInputs, setRegisterInputs] = useState({});
+
+  const [errorsInput, setErrorsInput] = useState({});
+
+  const [{ isLoading, response, errorsResponse }, doFetch] = useFetch(oneClass);
+  const [, setUser] = useLocalStorage("user", { isLogin, isChecking });
+  const [isSuccess, setIsSuccess] = useState(false);
+  const [isSingIn, setIsSingIn] = useState(false);
+
+  const dataInputsState = isLogin ? loginInputs : registerInputs;
+  const setDataInputsState = isLogin ? setLoginInputs : setRegisterInputs;
+
+  useEffect(() => {
+    setIsSuccess(false);
+    setReadyToEnter(false);
+    setIsChecking(false);
+    setDataInputsState({});
+    setErrorsInput({});
+  }, [isLogin, setDataInputsState]);
+
+  useEffect(() => {
+    if (0 !== Object.keys(errorsInput).length) {
+      setReadyToEnter(false);
+      return;
+    }
+    const check = false;
+    if (
```

Comment:
You should fix formatting here. If conditions if so long - create new function for checking them all.